test/contract-mock.js

```diff
@@ -22,42 +21,40 @@ module.exports = class MockContract {
 
   run(methodCall, methodParams) {
     switch (methodCall) {
-      case '01ffc9a7':
-        return this._01ffc9a7(`0x${methodParams.substring(0, 4 * 2)}`);
-      case 'd202158d':
-        return this._d202158d(`0x${methodParams.substring(0, 32 * 2)}`);
+      case '1626ba7e':
+        const hash = `0x${methodParams.substring(0, 32 * 2)}`;
+        const signatureLength = parseInt(
+          methodParams.substring(95 * 2, 96 * 2),
```

Comment:
What is the meaning of `95 * 2` and `96 * 2`? Should we assign them to constants so they don't seem like magic numbers?

Reply:
Fair point
This is basically saying "give me a byte slice from byte 95 to byte 96". The `* 2` part is because it's all a hex encoded string. 95 is kind of a magic number since it's how the params get encoded given 1271 contract ABI. I can move 0, and 95 to constants. I'll also give another look at web3 if they expose some of the ABI decoding magic.

Reply:
Ok cool. If it's universally understood then maybe its not necessary, I'm just not familiar with it.